vulnerabilities/importers/ruby.py

```diff
@@ -7,132 +7,188 @@
+        if not engine:
+            logger.error("Invalid engine name")
+        else:
+            purl = PackageURL(type="ruby", name=engine)
+            return AdvisoryData(
+                aliases=get_aliases(record),
+                summary=get_summary(record),
+                affected_packages=get_affected_packages(record, purl),
+                references=get_references(record),
+                date_published=get_publish_time(record),
+                url=advisory_url,
             )
-            for version in safe_versions
-        ]
 
-        references = []
-        if record.get("url"):
-            references.append(Reference(url=record.get("url")))
 
-        return AdvisoryData(
-            summary=record.get("description", ""),
-            affected_packages=nearest_patched_package(impacted_purls, resolved_purls),
-            references=references,
-            vulnerability_id=cve_id,
+def get_affected_packages(record, purl):
+    """
+    Return AffectedPackage objects one for each affected_version_range and invert the safe_version_ranges
+    ( patched_versions , unaffected_versions ) then passing the purl and the inverted safe_version_range
+    to the AffectedPackage object
+    """
+    safe_version_ranges = record.get("patched_versions", [])
+    # this case happens when the advisory contain only 'patched_versions' field
+    # and it has value None(i.e it is empty :( ).
+    if not safe_version_ranges:
+        safe_version_ranges = []
+    safe_version_ranges += record.get("unaffected_versions", [])
+    safe_version_ranges = [i for i in safe_version_ranges if i]
+
+    affected_packages = []
+    affected_version_ranges = [
+        GemVersionRange.from_native(elem).invert() for elem in safe_version_ranges
+    ]
+
+    for affected_version_range in affected_version_ranges:
+        affected_packages.append(
+            AffectedPackage(
+                package=purl,
+                affected_version_range=affected_version_range,
+            )
         )
+    return affected_packages
+
+
+def get_aliases(record) -> [str]:
+    aliases = []
+    if record.get("cve"):
+        aliases.append("CVE-{}".format(record.get("cve")))
+    if record.get("osvdb"):
+        aliases.append("OSV-{}".format(record.get("osvdb")))
+    if record.get("ghsa"):
+        aliases.append("GHSA-{}".format(record.get("ghsa")))
+    return aliases
 
-    @staticmethod
-    def categorize_versions(all_versions, unaffected_version_ranges):
 
-        for id, elem in enumerate(unaffected_version_ranges):
-            unaffected_version_ranges[id] = VersionRange.from_scheme_version_spec_string(
-                "semver", elem
+def get_references(record) -> [Reference]:
+    references = []
+    cvss_v2 = record.get("cvss_v2")
+    cvss_v3 = record.get("cvss_v3")
+
+    if record.get("url"):
+        if not (cvss_v2 or cvss_v3):
+            references.append(Reference(url=record.get("url")))
+        if cvss_v2:
```

Comment:
Do we want to keep CVSSV2 ? 

cc @pombredanne @DennisClark

Reply:
Done